standard/structs.md

```diff
@@ -476,3 +501,71 @@ Static constructors for structs follow most of the same rules as for classes. Th
+
+### §safe-scope-rules Safe scope constraint for `ref struct` types
+
+#### safe-scope-rules-general General
+
+At compile-time, each expression whose type is a ref struct is associated with a scope where that instance is safe, its ***safe-scope***. The safe-scope is a scope, enclosing an expression, to which it is safe for the value to escape to. If that scope is the entire function member, we say that the value is ***safe-to-return*** from the function member.
+
+The safe-scope records which scope a ref struct may be copied into. Given an assignment from an expression `E1` with a safe-scope `S1`, to an expression `E2` with safe-scope `S2`, it is an error if `S2` is a wider scope than `S1`.
+
+There are three different safe-scope values, the same as the ref-safe-scope values defined for reference variables (§ref-span-safety-escape-scopes): block, function-member, and calling-method. An expression `e1` of a ref struct type is constrained by its safe-scope as follows:
+
+- For a return statement `return e1`, the safe-scope of `e1` must be calling-method.
+- For an assignment `e1 = e2` the safe-scope of `e2` must be at least as wide a scope as the safe-scope of `e1`.
+
+For a method invocation if there is a `ref` or `out` argument of a `ref struct` type (including the receiver unless the type is `readonly`), with safe-scope `E1`, then no argument (including the receiver) may have a narrower safe-scope than `E1`.
+
+Any expression whose compile-time type is not a ref struct has a safe-scope of calling-method.
+
+A `default` expression, for any type, has safe-scope of calling-method.
+
+#### safe-scope-rules-parameter Parameter safe scope
+
+A formal parameter of a ref struct type, including the `this` parameter of an instance method, has a safe-scope of calling-method.
+
+#### safe-scope-rules-local Local variable safe scope
+
+A local variable of a ref struct type has a safe-scope as follows:
+
+- If the variable is an iteration variable of a `foreach` loop, then the variable's safe-scope is the same as the safe-scope of the `foreach` loop's expression.
+- A local of `ref struct` type uninitialized at the point of declaration has a safe-scope of calling method.
+- Otherwise the variable's declaration requires an initializer. The variable's safe-scope is the same as the safe-scope of its initializer.
+
+#### safe-scope-rules-field Field safe scope
+
+A a reference to a field, `e.F` where `F` is a ref struct type, has a safe-scope that is the same as the safe-scope of `e`.
+
+#### safe-scope-rules-operator Operators
+
+The application of a user-defined operator is treated as a method invocation (§safe-scope-method-invocation).
+
+For an operator that yields a value, such as `e1 + e2` or `c ? e1 : e2`, the safe-scope of the result is the narrowest scope among the safe-scopes of the operands of the operator. As a consequence, for a unary operator that yields a value, such as `+e`, the safe-scope of the result is the safe-scope of the operand.
```

Comment:
Is that the case for the conditional operator? I wouldn't expect the first operand to be relevant there, but I could be wrong.

Reply:
it doesn't matter, but it doesn't hurt either. The first operand must be a `bool`. Its safe scope is calling-method.  I couldn't wordsmith language that read reasonably well. Perhaps a note on the conditional operator's first operand?

Reply:
It is the safe-scope *of the result* so the first operand (`c`) is not relevant *per se* as the safe-scope is determined statically and not based on which alternative is selected at runtime. I wouldn't wordsmith this into the normative text, but a note could be added if desired.